src/codeu/chat/server/Server.java

```diff
@@ -140,7 +145,7 @@ private boolean onMessage(InputStream in, OutputStream out) throws IOException {
       final Uuid author = Uuid.SERIALIZER.read(in);
 	  final Uuid token = Uuid.SERIALIZER.read(in);
       final Uuid conversation = Uuid.SERIALIZER.read(in);
-      final String content = Serializers.STRING.read(in);
+      final String content = Serializers.readStringEnc(in, getServerPrivateKey());
```

Comment:
I'm not really sure passing the key to the serializer for encryption is such a good design. It introduces a tight coupling.

IMO Encryption should not be coupled with Serialization. They should be independent steps. You should take the byte stream output of serialization and then encrypt it. On the other side, decrypt the incoming byte stream and then deserialize it.